lib/matplotlib/__init__.py

```diff
@@ -1168,29 +1168,29 @@ def _init_tests():
             "Freetype build type is {}local".format(
                 "" if ft2font.__freetype_build_type__ == 'local' else "not "))
 
-    try:
-        import pytest
-    except ImportError:
-        print("matplotlib.test requires pytest to run.")
-        raise
-
 
 @cbook._delete_parameter("3.2", "switch_backend_warn")
 @cbook._delete_parameter("3.3", "recursionlimit")
 def test(verbosity=None, coverage=False, switch_backend_warn=True,
          recursionlimit=0, **kwargs):
     """Run the matplotlib test suite."""
-    _init_tests()
```

Comment:
But now you don't make the warning about FreeType. Why did the `pytest` import check have to move into this function and not just stay in `_init_tests`?

Reply:
You do still get the import warning about FreeType because the pytest configure calls `_init_tests`

https://github.com/matplotlib/matplotlib/blob/348e9feae16a9c5b3006955570f353fdb8058e54/lib/matplotlib/testing/conftest.py#L25

As originally written the `_init_tests` will be called twice: Once before the logger is set up (here), and then again by pytest (linked above) which by then logging is being captured. I didn't believe `_init_tests` needed to be called here because it doesn't raise an exception and you're not going to see the logs since they aren't set up at this point. If the warning in `_init_tests` isn't actually a warning it should be emitted in a way that a user can see it (currently it isn't).

The reason I moved the `pytest` import into this function is so this function doesn't need to blindly capture `ImportError` and can `return -1` for `pytest` not being installed. This is all detailed in the PR description if you would like to read that.

Reply:
@QuLogic thoughts?

Reply:
@QuLogic Does this satisfy your question? I've put this on the call agenda for tomorrow, but if you're ok with this, we don't have to spend time on it.